### ISSUE TITLE
inventory_overlay is ignored when item does not define inventory_image

### ISSUE THREAD

**Issue by ClobberXD** [minetest/minetest#8430]
##### Issue type

 - Bug report
 
##### Minetest version

```
5.1.0-dev-426bdba7
```

##### Summary

The itemdef field `inventory_overlay` is ignored (i.e. overlay isn't visible) when the item uses `minetest.inventorycube`. This bug is present in `0.4.x` as well.

Context: CTF duplicates all nodes from default, and sets `groups = {immortal = 1}` to make them indestructible, and I'm trying to set an inventory overlay to distinguish indestructible nodes from their normal counterparts. See the relevant bit of code here: https://github.com/ClobberXD/capturetheflag/blob/2884b8046fe8a6993a20e0dfa74461ef222ccb34/mods/ctf/ctf_map/nodes.lua#L768-L775

![screenshot_20190329_134038](https://user-images.githubusercontent.com/36130650/55218488-487f9100-5228-11e9-965e-e451b1f2e774.png)

Both the stacks in the hotbar have the `inventory_overlay` field defined. But as seen in the screenshot, the stack of `ctf_map:snow` displays the overlay correctly, whereas the stack of `ctf_map:stone` doesn't.

##### Steps to reproduce

- Set the `inventory_overlay` itemdef field to a custom image for two items: one that uses a single .png `inventory_image`, and one that uses `minetest.inventorycube` (most nodes from `default` fall under the latter).
- Obtain the items with the `inventory_overlay` field defined.
- Observe that the overlay isn't visible for the item that uses `minetest.inventorycube`.

**Comment by DS-Minetest** [comment]
I can't reproduce.
I've tested it with this:
```lua
minetest.register_node("test_invcube:node", {
	description = "inventorycube node",
	tiles = {"unknown_item.png"},
	inventory_image = minetest.inventorycube("unknown_object.png",
			"unknown_node.png", "unknown_object.png"),
	inventory_overlay = "camera_btn.png",
	groups = {oddly_breakable_by_hand = 3},
})

```

>one that uses `minetest.inventorycube` (most nodes from `default` fall under the latter).

This is wrong. They don't set an inventory image.

Searching through the ctf repository, I didn't find `minetest.inventorycube` being used more than two times (both not for stone).

I think that you think that setting no inventory image is equal to using `minetest.inventorycube`. But it istn't (enable `inventory_items_animations` (and restart minetest) to see the difference).

**However**, I can reproduce that `inventory_overlay` doesn't work for node items without inventory image.

**Comment by ClobberXD** [comment]
Apologies for assuming that `tiles` uses `minetest.inventorycube` internally.

**Comment by DS-Minetest** [comment]
I've made a PR to fix this: #8433
